### ISSUE TITLE
Make Alt/Cmd-Up go to the parent directory

### ISSUE THREAD

**Issue by mherrmann** [fman-community/fman#329]
Requested by @ar-jan in #35.

**Comment by crimoniv** [comment]
Hi @mherrmann and @ar-jan,

FWIW, I'm currently using the following **key bindings** that mimics the default **Nautilus** behaviour:
```
[
    ...
    { "keys": ["Alt+Up"], "command": "go_up" },
    { "keys": ["Alt+Down"], "command": "go_back" },
    { "keys": ["Alt+Left"], "command": "go_back" },
    { "keys": ["Alt+Right"], "command": "go_forward" },
    ...
]
```

Cheers!

**Comment by mherrmann** [comment]
Hi @crimoniv, thanks for sharing! That's the perfect solution. The question is, should this feature be implemented at all, or would `Alt-Up/Down` clash with other shortcuts people might expect?

**Comment by raguay** [comment]
I suggest just make this as a plugin call Nautilus compatibility or something similar. Having it as a default might start to make the defaults too big. Especially, if there are other defaults for other file-managers that people want to add-on.

**Comment by mherrmann** [comment]
Thx for the suggestion @raguay. While I'm a big fan of minimalism, I'm not 100% sure in this case. For two reasons: First, it's trivial to implement (just like @crimoniv suggested). So it doesn't bloat fman in terms of size or code complexity. Second, at least `Alt/Cmd+Up` is a very standard shortcut that works on all OSs. `Alt/Cmd+Down` however does not work on Windows. Hm...

**Comment by crimoniv** [comment]
I think that such a decision —to use shortcuts *à la **Nautilus***— should not be part of the defaults as it is a matter of personal preference, but OTOH it seems such a common shortcut to me —as I'm really used to **Nautilus**/**Caja**/**Thunar**—.

Maybe it should be published as a plugin, as @raguay said, and/or mentioned in [Custom shortcuts | fman](https://fman.io/docs/custom-shortcuts) (although I won't complain if it is finally included in **fman** :P).

**Comment by raguay** [comment]
I wasn't saying code bloat, but hotkey bloat. I think it's best to keep the standard set that is common across all platforms and majority of the file-managers as the base, then others, specialized ones as addons. For example, Alt/Cmd+up/down shows a preview on my mac. It doesn't move up or down. Therefore, maybe that is a more specialized command than a standard type command.

**Comment by mherrmann** [comment]
@crimoniv but it wouldn't be "use Nautilus's shortcuts", would it? It would be more akin to "support a shortcut that works on all major OSs".

@raguay I see. But isn't Alt/Cmd+Up "common across all platforms"?

**Comment by mherrmann** [comment]
(@raguay less sure about Alt+Down, as it doesn't work on Windows)

**Comment by raguay** [comment]
I take that back, Cmd-Up does go to parent on both Finder, Commander One, and Path Finder. But Cmd-down launches the file in both Finder and Path Finder on my macos. Commander One goes to the last entry on the directory for Cmd-down. Therefore, the results are not universal, but special, right?

**Comment by mherrmann** [comment]
@raguay I'd say they're universal for Cmd-Up and special for Cmd-Down. That all three of the file managers you mentioned support `Cmd-Up` is another pretty strong sign (imho) that this shortcut really is very universal.

**Comment by mherrmann** [comment]
`Alt+Up` does not work in Total Commander though.

**Comment by crimoniv** [comment]
It seems that GNOME, Xfce, KDE and Windows's file explorers use similar shortcuts —excluding <kbd>Alt</kbd>+<kbd>↓</kbd>, which only GNOME supports—, cannot talk about Krusader/Finder/Commander though.

### Nautilus
Shortcut | Action
--- | ---
<kbd>Alt</kbd>+<kbd>↑</kbd> | Go up (parent dir)
<kbd>Alt</kbd>+<kbd>↓</kbd> | Go down (go to selected dir / open selected file)
<kbd>Alt</kbd>+<kbd>←</kbd> | Go back (previous location)
<kbd>Alt</kbd>+<kbd>→</kbd> | Go forward (undo go back)

### Thunar
Shortcut | Action
--- | ---
<kbd>Alt</kbd>+<kbd>↑</kbd> | Go up (parent dir)
<kbd>Alt</kbd>+<kbd>↓</kbd> | ---
<kbd>Alt</kbd>+<kbd>←</kbd> | Go back (previous location)
<kbd>Alt</kbd>+<kbd>→</kbd> | Go forward (undo go back)

### Dolphin
Shortcut | Action
--- | ---
<kbd>Alt</kbd>+<kbd>↑</kbd> | Go up (parent dir)
<kbd>Alt</kbd>+<kbd>↓</kbd> | ---
<kbd>Alt</kbd>+<kbd>←</kbd> | Go back (previous location)
<kbd>Alt</kbd>+<kbd>→</kbd> | Go forward (undo go back)

### Windows File Explorer
Shortcut | Action
--- | ---
<kbd>Alt</kbd>+<kbd>↑</kbd> | Go up (parent dir)
<kbd>Alt</kbd>+<kbd>↓</kbd> | ---
<kbd>Alt</kbd>+<kbd>←</kbd> | Go back (previous location)
<kbd>Alt</kbd>+<kbd>→</kbd> | Go forward (undo go back)

**Comment by mherrmann** [comment]
Wow, that is one nice analysis. Thank you @crimoniv!

fman already implements <kbd>Alt</kbd>+<kbd>←</kbd> and <kbd>Alt</kbd>+<kbd>→</kbd>.

I don't think it makes too much sense to spend a lot of time debating this issue. @ar-jan requested <kbd>Alt</kbd>+<kbd>↑</kbd>, @crimoniv uses it as well and it's a very ubiquitous shortcut. I'll implement it.

Regarding <kbd>Alt</kbd>+<kbd>↓</kbd>, I won't add this for now because it's less ubiquitous. If it's requested again, we can still add it later.

**Comment by mherrmann** [comment]
Released as part of [fman 0.9.2](https://fman.io/changelog#v0.9.2).
